hack/chains/deploy_local_rekor.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# A wrapper for deploying rekor into a local cluster.
+# Uses the deploy_local_rekor.tpl file, which has a placeholder for
+# {domain} which is replaced by getting the domain from the openshift
+# cluster. 
+
+# Make sure we have our template file
+if [[ ! -f $PWD/deploy_local_rekor.tpl ]];then
+  echo "Template file 'deploy_local_rekor.tpl' not found in this directory "; exit 1 
+fi
+
+# Test if we have the argocd cli command:
+if ! command -v argocd &> /dev/null; then
+  echo "The argocd CLI is not installed. Please install and try again."
+  echo "See https://argo-cd.readthedocs.io/en/stable/getting_started/#2-download-argo-cd-cli for details"
+  exit 1
+fi
+
+# Get our domain from the ingresses config of the cluster
+domain=$( kubectl get ingresses.config.openshift.io cluster -o json|jq -r ".spec.domain" )
+
+# Set the template input file
+input_file="$PWD/deploy_local_rekor.tpl"
+
+# Set the file we're writing our template output to
+# TODO: Determine if we even need a file or can just use string
+deployment_file=$(mktemp --tmpdir deploy_local_rekor_XXXXXXX.yaml)
+
+# If we're writing temp files, let's trap sigs and remove the file
+# when the script exits
+trap "rm -f $deployment_file" 0 2 3 15
+
+# sed is awesome...
+sed -e "s/\${domain}/$domain/" $input_file > $deployment_file
+
+# If we want to see what's being written to the deployment file
+if [[ $1 == "--verbose" ]]; then
+  cat $deployment_file
+fi
+
+# TODO: Ensure we're actually logged in before trying to execute argocd command
+argocd app create --grpc-web --upsert -f $deployment_file #2> /dev/null
+
+# Check the exit code from above. From what I can tell, exit code 20
+# corresponds to "you've not logged in yet". If we can figure out the
+# TODO above, then we might be able to skip this or do it better
+if [[ $(echo $?) == 20 ]]; then
+  echo "Please login to your cluster with argocd"
+  echo "See https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_login/ for details"
+  echo "Note: for CRC, use the --sso option"
+  exit 1
+fi
+
+# Let the user know we've created the rekor app, set expectations,
+# and give them a command they can run to test if it's up and
+# running.
+echo """
+Created application "rekor". Syncing may take a few moments.
+Once synced, you can test that rekor is running by issuing 
+the following command:
+
+curl https://rekor-server.$domain/api/v1/log
+"""
```

Comment:
In hack/bootstrap.sh there's some examples of waiting for a route. Could copy that technique if you wanna make it more user friendly.